bindgen/features.rs

```diff
@@ -4,270 +4,227 @@
+                    Self::Nightly => None
+                }
+            }
+
+            const fn stable_releases() -> [(Self, u64); [$($minor,)*].len()] {
+                [$((Self::$variant, $minor),)*]
             }
         }
 
-        impl FromStr for RustTarget {
-            type Err = io::Error;
+        #[cfg(feature = "__cli")]
+        /// Strings of allowed `RustTarget` values
+        pub const RUST_TARGET_STRINGS: &[&str] = &[$(concat!("1.", stringify!($minor)),)*];
 
-            /// Create a `RustTarget` from a string.
-            ///
-            /// * The stable/beta versions of Rust are of the form "1.0",
-            /// "1.19", etc.
-            /// * The nightly version should be specified with "nightly".
-            fn from_str(s: &str) -> Result<Self, Self::Err> {
-                match s.as_ref() {
-                    $(
-                        stringify!($value) => Ok(RustTarget::$release),
-                    )*
-                    _ => Err(
-                        io::Error::new(
-                            io::ErrorKind::InvalidInput,
-                            concat!(
-                                "Got an invalid rust target. Accepted values ",
-                                "are of the form ",
-                                "\"1.0\" or \"nightly\"."))),
-                }
-            }
+        #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+        pub(crate) struct RustFeatures {
+            $($(pub(crate) $feature: bool,)*)*
+            $(pub(crate) $nightly_feature: bool,)*
         }
 
-        impl From<RustTarget> for String {
+        impl From<RustTarget> for RustFeatures {
             fn from(target: RustTarget) -> Self {
-                match target {
-                    $(
-                        RustTarget::$release => stringify!($value),
-                    )*
-                }.into()
+                if target == RustTarget::Nightly {
+                    return Self {
+                        $($($feature: true,)*)*
+                        $($nightly_feature: true,)*
+                    };
+                }
+
+                let mut features = Self {
+                    $($($feature: false,)*)*
+                    $($nightly_feature: false,)*
+                };
+
+                $(if target >= RustTarget::$variant {
+                    $(features.$feature = true;)*
+                })*
+
+                features
             }
         }
-    }
+    };
 }
 
-/// Defines an array slice with all RustTarget values
-macro_rules! rust_target_values_def {
-    ( $( $( #[$attr:meta] )* => $release:ident => $value:expr; )* ) => {
-        /// Strings of allowed `RustTarget` values
-        pub static RUST_TARGET_STRINGS: &'static [&str] = &[
-            $(
-                stringify!($value),
-            )*
-        ];
+// NOTE(emilio): When adding or removing features here, make sure to add the stabilization PR
```

Comment:
```suggestion
// NOTE: When adding or removing features here, make sure to add the stabilization PR
```
note is general

Reply:
what would _removing features_ mean... is it about deprecated things

Reply:
It could happen that a unstable feature is removed from nightly because of reasons or that its behavior change in such a way that bindgen cannot use it for what was intended before.